lib/internal/http2/core.js

```diff
@@ -997,6 +997,24 @@ class Http2Session extends EventEmitter {
 
     process.nextTick(emit, this, 'timeout');
   }
+
+  ref() {
+    if (this[kSocket]) {
+      assert(this[kSocket]._handle);
+      if (typeof this[kSocket]._handle.ref === 'function') {
```

Comment:
Why not just `this[kSocket].ref`? That’s what’s being called anyway, right?

Reply:
For generic duplex streams `this[kSocket]` is a `JSStreamWrap` object, which doesn't have a `ref`/`unref` field.

Actually, this makes me think that `net.Socket` itself should be checking that `_handle.ref` is a function, rather than here. I'll add a trailing commit that does so, we can drop it if there is already a good rationale for not checking in `net`.